scrapylib/deltafetch.py

```diff
@@ -93,10 +94,12 @@ def process_spider_output(self, response, result, spider):
                 key = self._get_key(r)
                 if self.db.has_key(key):
                     spider.log("Ignoring already visited: %s" % r, level=log.INFO)
+                    self.stats.inc_value('deltafetch/skipped', spider=spider)
```

Comment:
Hey @redapple, @pallih

If someone overrides `from_crawler` method - there's a chance that object is still initialized as:

```python
o = cls(dir, reset)
```

which leaves `self.stats = None` and this (and next changed) line will fail with `AttributeError`. Code should either fail fast with meaningful error message if `stats` are not passed or there should be check for stats value before incrementing stats:

```python
if self.stats:
    self.stats.inc_value('deltafetch/skipped', spider=spider)
```